Src/Extern/openPMD-api/AMReX_PlotFileUtilOPENPMD.H

```diff
@@ -0,0 +1,433 @@
+
+    void SetStep(int ts);
+    void CloseStep(int ts);
+    void CloseHandler();
+
+    void WriteSingleLevel (//const std::string &plotfilename,
+                           const MultiFab &mf,
+                           const Vector<std::string> &varnames,
+                           const Geometry &geom,
+                           Real t,
+                           //int level_step,
+                           const std::string &ignored_versionName = "HyperCLaw-V1.1",
+                           const std::string &ignored_levelPrefix = "Level_",
+                           const std::string &ignored_mfPrefix = "Cell",
+                           const Vector<std::string>& ignored_extra_dirs = Vector<std::string>());
+
+     void WriteMultiLevel (
+                           //int nlevels, will all levels in mf & geom
+                           const Vector<const MultiFab*> &mf,
+                           const Vector<std::string> &varnames,
+                           const Vector<Geometry> &geom,
+                           Real time,
+                           //const Vector<int> &level_steps,
+                           const Vector<IntVect> &ref_ratio,
+                           const std::string &ignored_versionName = "HyperCLaw-V1.1",
+                           const std::string &ignored_levelPrefix = "Level_",
+                           const std::string &ignored_mfPrefix = "Cell",
+                           const Vector<std::string>& ignored_extra_dirs = Vector<std::string>());
+
+
+    ////////////////////////////
+    struct AMReX_VarNameParser
+    {
+      AMReX_VarNameParser(std::string const& varname);
+      virtual ~AMReX_VarNameParser () = default;
+      AMReX_VarNameParser (AMReX_VarNameParser const&) = default;
+      AMReX_VarNameParser (AMReX_VarNameParser &&) = delete;
+      AMReX_VarNameParser& operator= (AMReX_VarNameParser const&) = default;
+      AMReX_VarNameParser& operator= (AMReX_VarNameParser &&) = delete;
+
+      void GetFieldNameModeInt(const std::string& varname);
+      void GetMeshCompNames (int meshLevel);
+
+      [[nodiscard]] std::vector< std::string > getFieldComponentLabels () const
+      {
+        using vs = std::vector< std::string >;
+        if (m_ThetaMode)
+          {
+            // if we write individual modes
+            vs fieldComponents{"r", "t", "z"};
+            return fieldComponents;
+          }
+        else
+          {
+            // if we just write reconstructed fields at theta=0 or are Cartesian
+            // note: 1D3V and 2D3V simulations still have 3 components for the fields
+            vs fieldComponents{"x", "y", "z"};
+            return fieldComponents;
+          }
+      }
+
+      [[nodiscard]] virtual std::vector< std::string >
+      getFieldAxisLabels () const
+      {
+        using vs = std::vector< std::string >;
+
+        // temporary resolution
+#if AMREX_SPACEDIM == 1
+        vs const axisLabels{"z"};
```

Comment:
I understand this is a virtual function. But I am wondering if it would be less confusing to simply use `AMREX_D_DECL("x","y","z")` than the WarpX convention.

Reply:
Sure, I can use AMREX_D_DECL("x","y","z") as default.